### PR TITLE
Added way to clone data source

### PR DESCRIPTION
**Pull request by kkrampa**
@emord @nickpell 
It might be useful e.g. when we want to create data source based on static data source.

### CODE REVIEW

**Review by nickpell — APPROVED** [review]

**Comment by nickpell** [comment]
How would I access this view?  Would I have to enter the url with the config id of the datasource I want to copy?  If so, is there a way to hook this up into a UI?

**Comment by kkrampa** [comment]
@nickpell updated

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -1110,6 +1110,37 @@ class CreateDataSourceView(BaseEditDataSourceView):
...
+    def page_url(self):
+        return reverse(self.urlname, kwargs=dict(domain=self.domain, config_id_to_clone=self.config_id_to_clone))
+
+    @property
+    def config(self):
+        data_source_to_clone = get_datasource_config_or_404(self.config_id_to_clone, self.domain)[0]
+        return DataSourceConfiguration(
+            domain=self.domain,
+            referenced_doc_type=data_source_to_clone.referenced_doc_type,
+            table_id=data_source_to_clone.table_id,
```
Comment (nickpell) [inline corehq/apps/userreports/views.py]: will the new data source point to the same table?  If so that seems problematic.

Comment (kkrampa) [inline corehq/apps/userreports/views.py]: ahh, you're right. That's an obvious mistake

**Comment by kkrampa** [comment]
Updated

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -1110,6 +1110,36 @@ class CreateDataSourceView(BaseEditDataSourceView):
...
+            configured_filter=data_source_to_clone.configured_filter,
+            configured_indicators=data_source_to_clone.configured_indicators,
+            named_expressions=data_source_to_clone.named_expressions,
+            named_filters=data_source_to_clone.named_filters,
+            display_name='{} - cloned {}'.format(
+                data_source_to_clone.display_name,
+                datetime.datetime.utcnow().strftime('%Y-%m-%d')
+            ),
+            backend_id=data_source_to_clone.backend_id
+        )
```
Comment (snopoke) [inline corehq/apps/userreports/views.py]: the better approach to cloning json objects is to do this `new = DataSourceConfiguration.wrap(old.to_json())`

and then change any values on `new` that need changing.

**Review by snopoke — COMMENTED** [review]
Not really sure about the need for this. @kkrampa is this something you've needed in the past?

I'm cautious to add functionality because it 'might' be useful in the future.

**Pull request closed without merging**